recipes/openassetio/all/conanfile.py

```diff
@@ -0,0 +1,177 @@
+import pathlib
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.apple import is_apple_os
+from conan.tools.files import apply_conandata_patches, get, copy, rm
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "openassetio"
+    description = "An open-source interoperability standard for tools and content management systems used in media production."
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/OpenAssetIO/OpenAssetIO"
+    topics = ("asset-pipeline", "vfx", "cg", "assetmanager", "vfx-pipeline")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "without_python": [True, False],
+        "python_version": ["3.7.12", "3.8.12", "3.9.7", "3.10.0"]
+    }
+    default_options = {
+        "shared": False,
+        "without_python": False,
+        "python_version": "3.9.7"
+    }
+    short_paths = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "9",
+            "clang": "12",
+            "apple-clang": "12",
+        }
+
+    def configure(self):
+        if self.options.without_python:
+            self.options.rm_safe("python_version")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("tomlplusplus/3.2.0")
+        if not self.options.without_python:
+            self.requires("openssl/1.1.1s")
+            self.requires(f"cpython/{self.options.python_version}")
+            if is_msvc(self):
+                # Required to create import .lib for building extension module.
+                self.options["cpython"].shared = True
```

Comment:
This was a fun one to debug. The error I kept seeing on Windows was good 'ole `undefined reference` to various Python symbols when building the Python extension module, yet `python39.lib` was clearly shown on the linker command-line.  

To re-iterate the commit message (84f70e90):

> In the static build of cpython the `python39.lib` file is, understandably, a static library rather than a Windows dll import
library.
> 
> When building a Python extension module we require the import library, since the extension module itself should not statically link the whole of Python(!).

It's still a little mysterious, though. I didn't dig into exactly why I saw `undefined reference` errors, since, although wrong, statically linking Python should have made its symbols available.  

So I could be mistaken in my conclusions.  Still, enforcing a shared `cpython` build on Windows made the build/tests work.

I can do some more digging if you feel it's worthwhile?